doc/programming/language/modules.rst

```diff
@@ -66,3 +69,18 @@ implicitly public.
 
 Spicy comes with a set of :ref:`library modules <library>` that you
 may import in your code to gain access to their functionality.
+
+Global Properties
+-----------------
+
+A module may define the following global properties:
+
+    ``%byte-order = ORDER;``
+        Defaults the byte order for any parsing inside this module to
+        `<expr>`, where ``ORDER`` must be of typei is type
+        :ref:`spicy_ByteOrder`.
+
+    ``%spicy-version = "VERSION";``
+        Specifies that the module requires a given minimum version of
+        Spicy, where ``VERSION`` must be a string of the form ``X.Y``
+        or ``X.Y.Z``.
```

Comment:
Do you think it would make sense to also support just `X`? This would become interesting especially once we release v1, v2 and so on.

Reply:
`%spicy-version = "1"` looks at bit odd to me, semantics aren't clear immediately. And one can express the same through `%spicy-version = "1.0"`

Reply:
As discussed offline, `"1"` would have the same implicit trailing zero semantics as e.g., `"1.1"`; `"1"` would require `>=1.0.0` like `"1.1" would require `>=1.1.0`. Let either allow single digit versions as well or always require all three digits.

Reply:
I favor with the whole specificity. New syntactic features are introduced at a specific version, best-specified with the whole specificity, not "this specific number is not good enough, but you're good-to-go as long as you are later than this". It would be a boon if what a user will see when their local Spicy is out-of-date with respect to that unsupported syntax, is an explicit message about the required minimum version.

Reply:
OTOH, if two different syntactic supports are needed, the value has to be the larger of the two. Probably conventional usage over time in this case will shift to overly conservatively putting a number that is not so precise, 
perhaps the next-higher minor, to capture something the author will think is some kind of "margin of safety".

Reply:
Folks, I'll leave this as it is for now, don't think it's worth iterating further at the moment.  If we want some other semantics, we can discuss, but for now "1.0" is just the same as "1.0.0", seems clear enough to me, and fits with semantic versioning as well where the patch level would usually not matter (i.e., we wouldn't introduce new features at that level). Duffy, the error messages includes what's required.